### PR TITLE
refactor(infra): tsify infra/elasticsearch with standalone pipeline

### PR DESCRIPTION
**Pull request by douglasduteil**

### CODE REVIEW

File: infra/elasticsearch/src/vectorizer/index.ts [hunk 1]
```diff
@@ -1,22 +1,34 @@
...
+  if (response.ok) {
+    const json: { outputs: number[][] } = await response.json();
+    return json.outputs;
+  } else {
+    throw new Error(response.statusText);
+  }
+}
+
+export function preprocess(text: string): string {
+  const stopWords = new Set(stopwords.map(stripAccents));
```
Comment (lionelB) [inline infra/elasticsearch/src/vectorizer/index.ts]: this should be outside of the function there is no need to recreate the stopwords list

File: lerna.json [hunk 2]
```diff
@@ -7,9 +7,6 @@
     }
   },
   "npmClient": "yarn",
-  "packages": [
-    "packages/*"
```
Comment (lionelB) [inline lerna.json:11]: is it ok to remove les package path for lerna ?

File: infra/elasticsearch/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+//
```
Comment (lionelB) [inline infra/elasticsearch/src/index.ts]: not needed

File: infra/elasticsearch/src/client.ts [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    throw error;
+  }
+}
+
+export async function indexDocumentsBatched({
+  client,
+  indexName,
+  documents,
+  size = 1000,
+}: UtilsOptions & { documents: Document[]; size: number }): Promise<void> {
```
Comment (lionelB) [inline infra/elasticsearch/src/client.ts:121]: should it be it simpler to split the arguments ?

File: infra/elasticsearch/src/client.ts [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+  const pIndicesToDelete = IndicesToDelete.map(async ({ index }) =>
+    client.indices.delete({ index })
+  );
+
+  return Promise.all(pIndicesToDelete).then(() => {
+    logger.info(`Remove ${pIndicesToDelete.length} old indices`);
+  });
+}
+
+export function* chunks<T = unknown>(
```
Comment (lionelB) [inline infra/elasticsearch/src/client.ts:147]: is ti a best practice to give a Generic an `unknonw` type ?

File: infra/elasticsearch/src/vectorizer/index.ts (hunk 1 above)
Comment (douglasduteil) [reply infra/elasticsearch/src/vectorizer/index.ts]: I would argue that less globales is better but ok.
I will restore the line 👍

File: lerna.json (hunk 2 above)
Comment (douglasduteil) [reply lerna.json:11]: lerna is directly using the workspace described in the `package.json`
This change could be a separate pr

File: infra/elasticsearch/src/index.ts (hunk 3 above)
Comment (douglasduteil) [reply infra/elasticsearch/src/index.ts]: my aesthetic habits ... 😏

File: infra/elasticsearch/src/client.ts (hunk 4 above)
Comment (douglasduteil) [reply infra/elasticsearch/src/client.ts:121]: What do you mean ?

File: infra/elasticsearch/src/client.ts (hunk 5 above)
Comment (douglasduteil) [reply infra/elasticsearch/src/client.ts:147]: it's more like a temporally workaround seems I do not know what is the type to put heer

**Comment by m-maillot** [comment]
Can we close it? `infra/elasticsearch` has been moved to `cdtn-admin` project which is currently move to TS.
@douglasduteil @lionelB

**Comment by douglasduteil** [comment]
Yeap @m-maillot

**Pull request closed without merging**